crates/host/src/nats/builder.rs

```diff
@@ -0,0 +1,251 @@
+//! An opinionated [crate::wasmbus::HostBuilder] that uses NATS as the primary transport.
+
+use std::{
+    collections::{BTreeMap, HashMap},
+    sync::Arc,
+    time::Duration,
+};
+
+use anyhow::{ensure, Context as _};
+use async_nats::{jetstream::kv::Store, Client};
+use nkeys::KeyPair;
+use serde::Deserialize;
+use serde_json::json;
+use tracing::{debug, error, instrument};
+use wasmcloud_control_interface::RegistryCredential;
+use wasmcloud_core::RegistryConfig;
+
+use crate::{
+    event::EventPublisher,
+    nats::{event::NatsEventPublisher, policy::NatsPolicyManager, secrets::NatsSecretsManager},
+    oci,
+    registry::{merge_registry_config, RegistryCredentialExt as _, SupplementalConfig},
+    secrets::SecretsManager,
+    store::StoreManager,
+    wasmbus::{config::BundleGenerator, HostBuilder},
+    PolicyHostInfo, PolicyManager, WasmbusHostConfig,
+};
+
+const DEFAULT_CTL_TOPIC_PREFIX: &str = "wasmbus.ctl";
+
+use super::{create_bucket, ctl::NatsControlInterfaceServer};
+
+/// Opinionated [crate::wasmbus::HostBuilder] that uses NATS as the primary transport and implementations
+/// for the [crate::wasmbus::Host] extension traits.
+///
+/// This builder is used to create a [crate::wasmbus::HostBuilder] and a [NatsControlInterfaceServer] for
+/// listening for control messages on the NATS message bus. Incoming messages will use the
+/// [crate::wasmbus::ctl::ControlInterfaceServer] trait to handle the messages and
+/// send them to the host.
+pub struct NatsHostBuilder {
+    // Required fields
+    ctl_nats: Client,
+    ctl_topic_prefix: String,
+    lattice: String,
+    config_generator: BundleGenerator,
+    registry_config: HashMap<String, RegistryConfig>,
+    enable_component_auction: bool,
+    enable_provider_auction: bool,
+
+    // Trait implementations for NATS
+    config_store: Arc<dyn StoreManager>,
+    data_store: Store,
+    policy_manager: Option<Arc<dyn PolicyManager>>,
+    secrets_manager: Option<Arc<dyn SecretsManager>>,
+    event_publisher: Option<Arc<dyn EventPublisher>>,
+}
+
+impl NatsHostBuilder {
+    /// Initialize the host with the NATS control interface connection
+    #[allow(clippy::too_many_arguments)]
+    pub async fn new(
```

Comment:
Nit: I think the cleaner API here (since you're already using a builder) would be set all of the defaults in the builder and have `default` and `new` be the same thing. `build` can take the NATS client as a required argument and all the other items can have methods for setting them